docs/using-react-redux/annotating-connected-components-with-flow.md

```diff
@@ -0,0 +1,255 @@
+
+In _component definition_, annotate the props with `Props`. The component will have access to all the injected props from `connect`:
+
+```jsx
+import * as React from 'react'
+
+const MyComponent = (props: Props) => (
+  <div onClick={props.dispatch1}>
+    {props.passthrough}
+    {props.fromMapStateToProps}
+  </div>
+)
+```
+
+When we export, this is also when we normally call `connect`, annotate the exported component with _just_ `OwnProps`:
+
+```jsx
+import * as React from 'react'
+
+// const MyComponent = ...
+
+export default (connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(MyComponent): React.AbstractComponent<OwnProps>)
+```
+
+### Annotating by providing explicit type parameters
+
+We may also annotate connected components by providing explicit type parameters at call of `connect` with the help of the [newest Flow Typed library definition for React Redux](https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/react-redux_v5.x.x/flow_v0.89.x-/react-redux_v5.x.x.js). Note that this will also require Flow v0.89+.
+
+The Flow Typed library definition declares `connect` as follows:
+
+```js
+declare export function connect<-P, -OP, -SP, -DP, -S, -D>(
+  mapStateToProps?: null | void,
+  mapDispatchToProps?: null | void,
+  mergeProps?: null | void,
+  options?: ?Options<S, OP, {||}, MergeOP<OP, D>>
+): Connector<P, OP, MergeOP<OP, D>>
+```
+
+The libdef also contains a [glossary of the abbreviations](https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/react-redux_v5.x.x/flow_v0.89.x-/react-redux_v5.x.x.js#L14) which decrypts the signature to:
+
+```jsx
+connect<Props, OwnProps, StateProps, DispatchProps, State, Dispatch>(…)
+```
+
+For the most common ways to connect components, we won't need all of the parameters. Normally, we need only `OwnProps` and `Props` at the call of `connect`, and `State` at the definition of `mapStateToProps`.
+
+We may use `_` ([what's this?](https://github.com/facebook/flow/commit/ec70da4510d3a092fa933081c083bd0e513d0518)) as placeholder at unused type parameter positions. A common `connect` call may look like this:
+
+```jsx
+connect<Props, OwnProps, _, _, _, _>(…)
+```
+
+We include examples for three major use cases of annotating `connect` with Flow:
+
+- Connecting stateless component with `mapStateToProps`
+- Connecting components with `mapDispatchToProps` of action creators
+- Connecting components with `mapStateToProps` and `mapDispatchToProps` of action creators
+
+#### Connecting stateless component with `mapStateToProps`
+
+```jsx
+type OwnProps = {|
+  passthrough: number,
+  forMapStateToProps: string,
+|};
+type Props = {|
+  ...OwnProps,
+  fromStateToProps: string
+|};
+const Com = (props: Props) => <div>{props.passthrough} {props.fromStateToProps}</div>
+
+type State = {a: number};
+type InputProps = {
+  forMapStateToProps: string
+};
+const mapStateToProps = (state: State, props: InputProps) => {
```

Comment:
InputProps also gets `passthrough`, so you should probably use `OwnProps` here and get rid of  `InputProps` altogether.

Reply:
I find it more straightforward to just use `OwnProps` in both places as well but I wasn't sure since the `connect_test` uses `InputProps`. Thanks for clarifying!

Reply:
Yeah, in a way the `InputProps` _is_ correct as long as it's not exact: It has forMapStateToProps _and potentially anything else_ too. But on the other hand as you already have `OwnProps` having all the props this receives why not using it here too? :)

Reply:
Sweet, thank you for the explanation!